src/Hl7.Fhir.Core/Model/ModelInfo.cs

```diff
@@ -412,6 +412,39 @@ public static bool IsProfiledQuantity(string type)
             return IsProfiledQuantity(definedType.Value);
         }
 
+        public static bool CheckMinorVersionCompatibility(string externalVersion)
+        {
+            if (string.IsNullOrEmpty(externalVersion))
+            {
+                return false;
+            }
+
+            var minorFhirVersion = GetMinorVersion(Version);
+            var externalMinorVersion = GetMinorVersion(externalVersion);
+
+            if (string.IsNullOrEmpty(minorFhirVersion) || string.IsNullOrEmpty(externalVersion))
+            {
+                return false;
+            }
+            else
+            {
+                return minorFhirVersion == externalMinorVersion;
+            }
+        }
+
+        public static string GetMinorVersion(string version)
+        {
+            var versionnumbers = version.Split('.');
+            if (versionnumbers != null && versionnumbers.Count() >= 2)
```

Comment:
`String.Split` will never return null.  Maybe you meant `version?.Split()` in the previous line?  Even so, I would guard for the `null` on the parameter, you don't need the null check here.